pkg/minishift/registration/redhat_test.go

```diff
@@ -112,22 +121,14 @@ func TestRedHatRegistratorUnregister(t *testing.T) {
 	}
 	commander := provision.GenericSSHCommander{Driver: d}
 	registrator := NewRedHatRegistrator(commander)
-	s.CommandToOutput["sudo subscription-manager version"] = `server type: This system is currently not registered.`
-	if err := registrator.Unregister(param); err != nil {
-		t.Fatal("Distribution shouldn't able to unregister")
-	} else {
-		cmd := "sudo subscription-manager unregister"
-		if _, ok := s.Commands[cmd]; ok{
-			t.Fatalf("Expected command: %s", cmd)
-		}
-	}
+
 	s.CommandToOutput["sudo subscription-manager version"] = `server type: RedHat Subscription Management`
 	if err := registrator.Unregister(param); err != nil {
-		t.Fatal("Distribution should able to unregister")
+		t.Fatal("Distribution should be able to unregister")
 	} else {
 		cmd := "sudo subscription-manager unregister"
 		if _, ok := s.Commands[cmd]; !ok{
-			t.Fatalf("Expected command: %s", cmd)
+			t.Fatal("Expected command: sudo subscription-manager unregister")
```

Comment:
> s.Commands contain all commands executed on the mock ssh server, 

Exactly, this is why your test is actually quite weak. Really a better test would assert the total number of executed commands (which we already could) and the order (which afaik we cannot do). Hence, I said once, that we should extend the mock ssh server. Different issue though. Happy to accept the "simpler" form of the test"

>  Now if we put cmd variable then error message would be misleading

No, how can it be? You use the value of _cmd_ as key into the commands map. If the command is there, you say the test passed, if not you say the test fails. Hence the expected commands for your test is whatever the value is in _cmd_.

> I got what is confusion here so I did some more modification.

You are making it worse.